File: routes/log.rb
# Handles logging the results of the user submissions. Should be merged into
#  the verification route when it gets created.

require 'sinatra'
require 'mongo'

# Log the data
post '/log' do
	data = JSON.parse(request.body.read)
	if (data.key?('type') and data.key?('module') and data.key?('name') and
		data.key?('author') and data.key?('code') and data.key?('time') and data.key?('correct'))

		data['timestamp'] = Time.now.utc.iso8601
		collection = settings.mongo[:data]
		collection.insert_one(data)
		status 200
	else
		status 400
	end
end

File: routes/log.js
var express = require('express')
var router = express.Router()

router.post('/log', (req, res) => {
	if (!(
		'type' in req.body &&
		'module' in req.body &&
		'name' in req.body &&
		'author' in req.body &&
		'code' in req.body &&
		'time' in req.body &&
		'correct' in req.body
	)) {
		res.status(400)
		res.send('Requires type, module, name, author, code, time, and correct fields.')
	} else {
		var time = new Date()
		req.body.timestamp = time.toISOString() // Note that this is an ISO 8601 timestamp, which does not account for time zone
		var data = req.db.collection('data')
		data.insertOne(req.body)
		res.json(req.body)
	}
})

module.exports = router
